akka-docs/src/main/paradox/scala/extending-akka.md

```diff
@@ -20,32 +20,62 @@ So let's create a sample extension that just lets us count the number of times s
 
 First, we define what our `Extension` should do:
 
-@@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension }
+Scala
+:  @@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension }
+
+Java
+:  @@snip [ExtensionDocTest.java]($code$/java/jdocs/extension/ExtensionDocTest.java) { #imports #extension }
 
 Then we need to create an `ExtensionId` for our extension so we can grab a hold of it.
 
-@@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extensionid }
+Scala
+:  @@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extensionid }
+
+Java
+:  @@snip [ExtensionDocTest.java]($code$/java/jdocs/extension/ExtensionDocTest.java) { #imports #extensionid }
 
 Wicked! Now all we need to do is to actually use it:
 
-@@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension-usage }
+Scala
+:  @@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension-usage }
+
+Java
+:  @@snip [ExtensionDocTest.java]($code$/java/jdocs/extension/ExtensionDocTest.java) { #extension-usage }
 
 Or from inside of an Akka Actor:
 
-@@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension-usage-actor }
+Scala
+:  @@snip [ExtensionDocSpec.scala]($code$/scala/docs/extension/ExtensionDocSpec.scala) { #extension-usage-actor }
+
+Java
+:  @@snip [ExtensionDocTest.java]($code$/java/jdocs/extension/ExtensionDocTest.java) { #extension-usage-actor }
+
+@@@ div { .group-scala }
 
```

Comment:
The below technique using trait is only applicable to Scala.
![image](https://user-images.githubusercontent.com/7414320/27123411-c1387ae2-5128-11e7-93c6-915ab6253d1d.png)